miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/AbstractST.java

```diff
@@ -647,4 +663,67 @@ protected static Integer readInteger(String resultJson, String path) {
   protected static long readLong(String resultJson, String path) {
     return readInteger(resultJson, path).longValue();
   }
+
+  /**
+   * Tests model "edit entity" form setup
+   * 
+   * @param url url to query
+   * @return ModelAndView from the endpoint
+   */
+  protected ModelAndView testModelFormSetup(String url)
+      throws Exception {
+    ResultActions ac = getMockMvc().perform(get(url)
+        .accept(MediaType.APPLICATION_JSON));
+
+    if (DEBUG_MODE)
+      ac = ac.andDo(print());
+
+    ModelAndView mvc = ac.andExpect(status().isOk())
+        .andReturn().getModelAndView();
+
+    return mvc;
+  }
+
+  /**
+   * Tests model "edit entity" form setup
+   * 
+   * @param url url to query
+   * @param listModelAttribute Model attribute, if relevant
+   * @return Resulting JSON response
+   */
+  protected String testModelFormSetup(String url, String listModelAttribute) throws Exception {
+    return testModelFormSetup(url).getModel().get(listModelAttribute).toString();
+  }
+
+  /**
+   * Tests model single create endpoints.
+   * 
+   * @param url
+   * @param title
+   * @throws Exception
+   */
+  protected ResultActions baseTestNewModel(String url, String title) throws Exception {
+    ResultActions ac = getMockMvc().perform(get(url).accept(MediaType.APPLICATION_JSON));
+    if (DEBUG_MODE)
+      ac = ac.andDo(print());
+
+    ac = ac.andExpect(status().isOk())
+        .andExpect(model().attribute("title", title));
+    return ac;
+  }
+
+  /**
+   * Tests single-entity edit model endpoints (usually a get mapping to /{entityId})
+   * 
+   * @param url URL to query
+   * @return ResultActions from request
+   */
+  protected ResultActions baseTestEditModel(String url) throws Exception {
```

Comment:
This is unused in this PR and is identical to `testModelFormSetup` except for returning ResultActions instead of the ModelAndView. `baseTestEditModel` is a better name though